src/wp-content/themes/twentyfifteen/js/functions.js

```diff
@@ -5,130 +5,162 @@
  * Contains handlers for navigation and widget area.
  */
 
-( function( $ ) {
-	var $body, $window, $sidebar, resizeTimer,
-		secondary, button;
+( function() {
+	function domReady( callback ) {
+		if ( document.readyState === 'complete' || document.readyState === 'interactive' ) {
+			callback();
+		} else {
+			document.addEventListener( 'DOMContentLoaded', callback );
+		}
+	}
+
+	function toggleClass( element, className ) {
+		if ( -1 !== element.className.indexOf( className ) ) {
+			element.className = element.className.replace( ' ' + className, '' );
+		} else {
+			element.className += ' ' + className;
+		}
+	}
+
+	var sidebar, resizeTimer, secondary, button;
 
 	function initMainNavigation( container ) {
+		if ( ! container ) {
+			return;
+		}
+
 		// Add dropdown toggle that display child menu items.
-		container.find( '.menu-item-has-children > a' ).after( '<button class="dropdown-toggle" aria-expanded="false">' + screenReaderText.expand + '</button>' );
+		Array.from( container.querySelectorAll( '.menu-item-has-children > a' ) ).forEach( function( menuLink ) {
+			var dropdownToggle = document.createElement( 'button' );
+			dropdownToggle.className = 'dropdown-toggle';
+			dropdownToggle.setAttribute( 'aria-expanded', 'false' );
+			dropdownToggle.innerHTML = screenReaderText.expand;
+			menuLink.parentElement.insertBefore( dropdownToggle, menuLink.nextElementSibling );
+		} );
 
 		// Toggle buttons and submenu items with active children menu items.
-		container.find( '.current-menu-ancestor > button' ).addClass( 'toggle-on' );
-		container.find( '.current-menu-ancestor > .sub-menu' ).addClass( 'toggled-on' );
-
-		container.find( '.dropdown-toggle' ).on( 'click', function( e ) {
-			var _this = $( this );
-			e.preventDefault();
-			_this.toggleClass( 'toggle-on' );
-			_this.next( '.children, .sub-menu' ).toggleClass( 'toggled-on' );
-			_this.attr( 'aria-expanded', _this.attr( 'aria-expanded' ) === 'false' ? 'true' : 'false' );
-			_this.html( _this.html() === screenReaderText.expand ? screenReaderText.collapse : screenReaderText.expand );
+		Array.from( container.querySelectorAll( '.current-menu-ancestor > button, .current-menu-ancestor > .sub-menu' ) ).forEach( function( subItem ) {
+			subItem.className += 'toggle-on';
+		} );
+
+		Array.from( container.querySelectorAll( '.dropdown-toggle' ) ).forEach( function( dropdownToggle ) {
+			dropdownToggle.addEventListener( 'click', function( e ) {
+				e.preventDefault();
+				toggleClass( this, 'toggle-on' );
```

Comment:
```suggestion
				dropdownToggle.classList.toggle( 'toggle-on' );
```

I'm not sure if `this` does the right thing inside of a `forEach` loop, but in any case it's a good idea to avoid `this` whenever possible.

Reply:
Hmm, not sure, `this` in an event listener on an element should always be the element, and I think that improves code readability over having to read `dropdownToggle` again every time. 🤔

Reply:
> `this` in an event listener on an element should always be the element

Not if it's already been bound to something else 😕 Try [this fiddle](https://jsfiddle.net/uwtz6bc3/) to see what I mean (be sure to open the fiddle's console on the bottom right).

In that example, you've got different values of `this` depending on whether you used a plain function or an arrow function, or whether you bound the plain function. This illustrates the possibility for someone to introduce a bug simply by changing the function to an arrow function, for example.

This is why a lot of JS developers are generally wary of `this`, and why I'd argue that it's much less readable than any well-named function parameter; because it can take on different values in ways that may require looking through the code very attentively.

Reply:
Here's [an aptly timed article](https://web.dev/javascript-this/) on all the mental work a developer needs to do in order to figure out what `this` might mean, in any given context.